doc/guide/r_quickstart.hpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file r_quickstart.hpp
+ * @author Yashwant Singh Parihar
+
+@page r_quickstart mlpack in R quickstart guide
+
+@section r_quickstart_intro Introduction
+
+This page describes how you can quickly get started using mlpack from R and
+gives a few examples of usage, and pointers to deeper documentation.
+
+This quickstart guide is also available for @ref python_quickstart "Python"
+@ref cli_quickstart "the command-line", @ref julia_quickstart "Julia" and
+@ref go_quickstart "Go".
+
+@section r_quickstart_install Installing mlpack
+
+Installing the mlpack bindings for R is straightforward; you can just use
+cran mirror:
+
+@code{.R}
+install.packages('mlpack')
+@endcode
+
+Building the R bindings from scratch is a little more in-depth, though.  For
+information on that, follow the instructions on the @ref build page, and be sure
+to specify @c -DBUILD_R_BINDINGS=ON to CMake; you may need to also set the
+location of the R program with @c -DR_EXECUTABLE=/path/to/R.
```

Comment:
So actually I'm not sure it's needed to have an `mlpack-r` repository.  Based on my understanding, an upload to CRAN happens not by, e.g., giving the location of a Github repository (like in Julia), but instead by directly uploading a `.tar.gz`.  This means that instead of an `mlpack-r` repository, we could do one of the following:

 (1) at every mlpack release, manually build with `make r`, package what's in `build/src/mlpack/bindings/r/` and upload to CRAN
 (2) whenever a new release is tagged, fire off a Travis or Azure job that does (1) automatically
 (3) same as (2) except use Jenkins
 (4) same as (2) except use Github Actions

Does that work?  Or will there be people who... don't want to use CRAN, but install directly from a Github URL or something?

Reply:
I guess we may require to do some discussion here. As per my knowledge its good do a `R CMD check` on the package in the "proper" environment, before releasing/publishing the package. If we do `R CMD check` using `ctest` while building with `mlpack`, it may be possible that for building the `RcppMLPACK` R may use installed boost/cereal serialization files required to build mlpack or it may use some mlpack files which might not correctly travelled to `RcppMLPACK` while cmaking. Hence I thought it may be helpful if we go for `mlpack-r` with a `R CMD check` in proper environment, Here Github Actions/Azure might help us with the package release after a successful build on`mlpack-r`.
Or it may be possible that we may do these checks locally after every release in the proper environment and then go for a CRAN upload.
Just an idea, I am really sorry if I am missing something or making some blunder here.

Reply:
:+1: thanks for the explanation @Yashwants19.  In that case, if we were using the `mlpack-r` approach you described, would the `mlpack-r` repository just contain a clone of the `mlpack` repository at the appropriate tag?

I do agree that having some kind of system set up to run `R CMD check` before submitting to CRAN would be great, although it seems like CRAN will automatically run `R CMD check` too---so maybe we can just depend on CRAN failing the uploaded package as our "CI check" for a new release?

Reply:
> so maybe we can just depend on CRAN failing

Please don't. CRAN really is just a few overworked humans.  It is not a test bed. We have _e.g._ [Rhub](https://builder.r-hub.io) for that.

Consider CRAN uploads as a manual, explicit step.  Think Journal submission (which is exaggerating somewhat to make the point), so "random copy of a file to Dropbox" or another web service.

Reply:
Eek, I didn't realize that.  Yes, in this case, let's not overwork the humans even more. :)